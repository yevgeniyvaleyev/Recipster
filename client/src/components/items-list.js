import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { List } from 'material-ui';
import { ItemDetailsShort } from './item-details-short';

export class ItemsList extends Component {

  render () {
    const { items } = this.props;
        
    return (
      <List>
        {items.map((item) => (
          <ItemDetailsShort key={item.recipe_id} item={item} />
        ))}
      </List>
    )
  }
}

ItemsList.propTypes = {
  items: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired
};