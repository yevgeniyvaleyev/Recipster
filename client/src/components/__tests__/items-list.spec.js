import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { ItemsList } from '../items-list';
import { BrowserRouter } from 'react-router-dom';

test('should render component', () => {
  const loading = false;
  const items = [{
    title: 'test title',
    image_url: 'test.png',
    publisher: 'test',
    recipe_id: 1
  }]

  const component = renderer.create(
    <MuiThemeProvider>
      <BrowserRouter>
        <ItemsList items={items} loading={loading} />
      </BrowserRouter>
    </MuiThemeProvider>,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

