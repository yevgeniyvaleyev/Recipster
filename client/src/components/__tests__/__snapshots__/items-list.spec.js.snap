// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<div
  style={
    Object {
      "muiPrepared": true,
      "padding": "8px 0px 8px 0px",
    }
  }
>
  <a
    aria-current={false}
    className="sc-bdVaJa llYZfh"
    href="/item/1"
    onClick={[Function]}
    style={undefined}
  >
    <div>
      <span
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        open={null}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "backgroundColor": null,
            "border": 10,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "fontWeight": "inherit",
            "lineHeight": "16px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "marginLeft": 0,
                "muiPrepared": true,
                "paddingBottom": 16,
                "paddingLeft": 72,
                "paddingRight": 16,
                "paddingTop": 20,
                "position": "relative",
              }
            }
          >
            <img
              alt="test title"
              className={undefined}
              size={40}
              src="test.png"
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgb(188, 188, 188)",
                  "borderRadius": "50%",
                  "color": "#ffffff",
                  "display": "inline-flex",
                  "fontSize": 20,
                  "height": 40,
                  "justifyContent": "center",
                  "left": 16,
                  "muiPrepared": true,
                  "position": "absolute",
                  "top": 16,
                  "userSelect": "none",
                  "width": 40,
                }
              }
            />
            <div
              style={
                Object {
                  "muiPrepared": true,
                }
              }
            >
              test title
            </div>
            <div
              style={
                Object {
                  "WebkitBoxOrient": null,
                  "WebkitLineClamp": null,
                  "color": "rgba(0, 0, 0, 0.54)",
                  "display": null,
                  "fontSize": 14,
                  "height": 16,
                  "lineHeight": "16px",
                  "margin": 0,
                  "marginTop": 4,
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            >
              test
            </div>
          </div>
        </div>
      </span>
    </div>
  </a>
</div>
`;
